barrels/LogMonkey/README.md

```diff
@@ -0,0 +1,77 @@
+# LogMonkey
+
+The LogMonkey barrel provides logging utilities to aid in developing and debugging Connect IQ applications.
+
+Information can be logged to certain log levels via the available APIs. The supported log levels are:
+| API             | Purpose                                                |
+| --------------- | ------------------------------------------------------ |
+| LogMonkey.Debug | General debugging information                          |
+| LogMonkey.Warn  | Warning messages about potential issues within the app |
+| LogMonkey.Error | Errors that have occurred within the app               |
+
+## Adding LogMonkey to Your Project
+
+See the general instructions for [including Monkey Barrels](https://github.com/garmin/connectiq-apps/tree/master/barrels#including-monkey-barrels).
+
+## Logging Information
+
+Once you've added LogMonkey to your project you can import it into your code via `using` statements.
+```
+using LogMonkey as Log;
+```
+Within you're project's code you can log messages, variables and Exceptions via the available APIs.
+```
+using LogMonkey as Log;
+
+class MyClassOfThings {
+
+    function aFunctionOfLogic(aParameter) {
+        Log.Debug.logMessage("MyClassOfThings", "aFunctionOfLogic()");
+        Log.Debug.logVariable("aFunctionOfLogic() param", "aParameter", aParameter);
+        ...
+        try {
+            ...do something horribly, horribly wrong...
+        } catch (ex) {
+            Log.Error.logException("Attempting to do XYZ", ex);
+        }
+        ...
+        if (aLocalVariable == aPotentiallyProblematicValue) {
+            Log.Warn.logVariable("aFunctionOfLogic()", "aLocalVariable", aLocalVariable);
+        }
+        ...
+    }
+
+}
+```
+
+## Viewing and Parsing Logs
+
+LogMonkey makes use of `Toybox.System.println()` calls to log information. This means that if your app also makes use of `pinrtln()` calls they will show up alongside the log entries.
+
+### Output Format
+
+LogMonkey logs each line in the file in the following format: `(log format version)[timestamp] {log level} tag: message`. Details for each block of this format are given in the table below.
+| Block              | Format                                                 | Description                                                                         |
+| ------------------ | ------------------------------------------------------ | ----------------------------------------------------------------------------------- |
+| log format version | A value beginning with "lfm" then a number             | The output format version of the line in the log.                                   |
+| timestamp          | MM-DD-YYYY HH:MM:SS                                    | A time stamp of when the log entry occurred.                                        |
+| log level          | Regular Expression: `[a-zA-Z0-9_]`                     | The priority level of the log entry. LogMonkey supports values of `D`, `W` and `E`. |
```

Comment:
Are log levels required to be a single alpha-numeric character or underscore? This does break into the question about whether the Logger class initialize method should be documented or if there should be an undocumented method for specifying the `level`.

Reply:
The log level value should can be more than a single character; the regular expression here should be updated to reflect that. I wanted the standard format to be meaningful but concise so it's less overwhelming to view a raw log file.